modmail/extensions/configuration_manager.py

```diff
@@ -0,0 +1,282 @@
+        """
+        Set the provided option to new_value.
+
+        Raises an UnableToModifyConfig error if there is an issue
+        """
+        if new_value in [marshmallow.missing, attr.NOTHING]:
+            raise UnableToModifyConfig(
+                f"`{option.rsplit('.', 1)[-1]}` is a required configuration variable and cannot be reset."
+            ) from None
+
+        try:
+            _recursive_setattr(self.bot.config.user, option, new_value)
+        except (attr.exceptions.FrozenAttributeError, attr.exceptions.FrozenInstanceError):
+            raise UnableToModifyConfig(
+                f"Unable to set `{option}` as it is frozen and cannot be edited during runtime."
+            ) from None
+        else:
+            return (option, new_value)
+
+    @config_group.command(name="set_default", aliases=("set-default",))
+    async def set_default(self, ctx: Context, option: KeyConverter) -> None:
+        """Reset the provided configuration value to the default."""
+        value = _recursive_getattr(self.bot.config.default, option)
+        await self.set_config_value(option, value)
+
+        await responses.send_positive_response(
+            ctx, f"Successfully set `{option}` to the default of `{value}`."
+        )
+
+    @config_group.command(name="set", aliases=("edit",))
+    async def modify_config_command(self, ctx: Context, option: KeyConverter, value: str) -> None:
+        """Modify an existing configuration value."""
+        metadata = self.config_fields[option]
+
+        if metadata.frozen:
+            await responses.send_negatory_response(
+                ctx, f"Unable to modify `{option}` as it is frozen and cannot be edited during runtime."
+            )
+            return
+
+        # since we are in the bot, we are able to run commands.run_converters()
+        # we have a context object, and several other objects, so this should be able to work
+        annotation = metadata.discord_converter or metadata.type
+        param = inspect.Parameter("value", 1, default=metadata.default, annotation=annotation)
+        converted_result = await commands.run_converters(ctx, annotation, value, param)
+
+        discord_converter_attribute = metadata.modmail_metadata.discord_converter_attribute
+        if isinstance(discord_converter_attribute, types.FunctionType):
+            converted_result = discord_converter_attribute(converted_result)
+
+        await self.set_config_value(option, value)
+
+        if converted_result == value:
+            response = f"Successfully set `{option}` to `{value}`."
+        else:
+            response = f"Successfully set `{option}` to `{converted_result}` (converted from `{value}`)"
+        await responses.send_positive_response(ctx, response)
+
+    @config_group.command(name="get", aliases=("show",))
+    async def get_config(self, ctx: Context, option: KeyConverter) -> None:
```

Comment:
`get` shouldn't work if the value is hidden, like in the case of `bot.token`

Reply:
Will fix when renovating the configuration manager.